tempodb/wal/append_block.go

```diff
@@ -238,3 +243,20 @@ func (a *AppendBlock) file() (*os.File, error) {
 
 	return a.readFile, err
 }
+
+func (a *AppendBlock) adjustTimeRangeForSlack(start uint32, end uint32, additionalStartSlack time.Duration) (uint32, uint32) {
+	now := time.Now()
+	startOfRange := uint32(now.Add(-a.ingestionSlack).Add(-additionalStartSlack).Unix())
+	endOfRange := uint32(now.Add(a.ingestionSlack).Unix())
+
+	if start < startOfRange {
+		metricWarnings.WithLabelValues(a.meta.TenantID, reasonOutsideIngestionSlack).Inc()
+		start = uint32(now.Unix())
```

Comment:
Should these clip to the range instead of now?

Reply:
I originally did that and moved away from it. Now that I'm thinking through it again I am in agreement.

Reply:
Oh, so the reason I decided to do use `time.Now()` is b/c we know we are regularly receiving backdated spans from multiple weeks ago. If we use `start` instead of `time.Now()` then every block's time range will always start `ingestion_time_range_slack` earlier than it actually does. Also, every replayed block's start time will always be replay time - `max_block_duration` - `ingestion_time_range_slack`.